vcd/datasource_vcd_catalog_media_test.go

```diff
@@ -0,0 +1,147 @@
+// +build catalog ALL functional
+
+package vcd
+
+import (
+	"fmt"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+// Test catalog and catalog media data sources
+// Using a catalog data source we reference a catalog media data source
+// Using a catalog media data source we create another catalog media
+// where the description is the first data source ID
+func TestAccVcdCatalogAndMediaDatasource(t *testing.T) {
+	var TestCatalogMediaDS = "TestCatalogMediaDS"
+	var TestAccVcdDataSourceMedia = "TestAccVcdCatalogMediaBasic"
+	var TestAccVcdDataSourceMediaDescription = "TestAccVcdCatalogMediaBasicDescription"
+
+	var media govcd.Media
+
+	var params = StringMap{
+		"Org":              testConfig.VCD.Org,
+		"Catalog":          testSuiteCatalogName,
+		"NewCatalogMedia":  TestCatalogMediaDS,
+		"OvaPath":          testConfig.Ova.OvaPath,
+		"UploadPieceSize":  testConfig.Ova.UploadPieceSize,
+		"UploadProgress":   testConfig.Ova.UploadProgress,
+		"Tags":             "catalog",
+		"CatalogMediaName": TestAccVcdDataSourceMedia,
+		"Description":      TestAccVcdDataSourceMediaDescription,
+		"MediaPath":        testConfig.Media.MediaPath,
+	}
+
+	configText := templateFill(testAccCheckVcdCatalogMediaDS, params)
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+
+	debugPrintf("#[DEBUG] CONFIGURATION: %s", configText)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { preRunChecks(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: catalogMediaDestroyed(testSuiteCatalogName, TestCatalogMediaDS),
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config:             configText,
+				ExpectNonEmptyPlan: true,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVcdCatalogMediaExists("vcd_catalog_media."+TestAccVcdDataSourceMedia, &media),
+					resource.TestMatchOutput("owner_name", regexp.MustCompile(`^\w*$`)),
+					resource.TestMatchOutput("creation_date", regexp.MustCompile(`^^\d{4}-\d{2}-\d{2}.*`)),
+					resource.TestCheckOutput("status", "RESOLVED"),
+					resource.TestMatchOutput("storage_profile_name", regexp.MustCompile(`(.|\s)*\S(.|\s)*`)),
```

Comment:
This regular expression will easily match "a" or "aaa" or " a ".
`(.|\s)*` == **zero* or more occurrence of either any character or a space. 
The only mandatory character is `\S` which means "a non-space" character. Is this what we are trying to catch?

Reply:
My aim is non empty string, I can't guess/verify anything else

Reply:
updated to `non-empty`